### PR TITLE
Remove submodules

### PR DESCRIPTION
**Pull request by adambabik**
At the current stage, submodules, especially for `protocol` and `eth-node` packages, bring more troubles than advantages.

The biggest problem is that we use `replace`s in the main `go.mod` which makes status-go act differently depending whether it is used as a main program vs a library in other projects.

### CODE REVIEW

File: protocol/contact.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ type Contact struct {
 	SystemTags []string `json:"systemTags"`
 
 	DeviceInfo    []ContactDeviceInfo `json:"deviceInfo"`
-	TributeToTalk string              `json:"tributeToTalk,omitEmpty"`
+	TributeToTalk string              `json:"tributeToTalk,omitempty"`
```
Comment (adambabik) [inline protocol/contact.go:61]: Potentially breaking change but tests pass. cc @cammellos

File: protocol/message.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ type Message struct {
 
 	// Replace indicates that this is a replacement of a message
 	// that has been updated
-	Replace   string           `json:"replace,omitEmpty"`
+	Replace   string           `json:"replace,omitempty"`
```
Comment (adambabik) [inline protocol/message.go:96]: Potentially breaking change cc @cammellos

File: protocol/message.go [hunk 3]
```diff
@@ -130,10 +129,10 @@ func (m *Message) MarshalJSON() ([]byte, error) {
 		ParsedText        json.RawMessage                  `json:"parsedText"`
 		LineCount         int                              `json:"lineCount"`
 		Text              string                           `json:"text"`
-		ChatId            string                           `json:"chatId"`
+		ChatID            string                           `json:"chatId"`
 		LocalChatID       string                           `json:"localChatId"`
 		Clock             uint64                           `json:"clock"`
-		Replace           string                           `json:"replace,omitEmpty"`
+		Replace           string                           `json:"replace"`
```
Comment (adambabik) [inline protocol/message.go:135]: A unit test was failing so I removed it. `omitEmpty` did not work, it is `omitempty`. cc @cammellos

File: protocol/messenger.go [hunk 4]
```diff
@@ -74,12 +74,12 @@ type RawResponse struct {
 }
 
 type MessengerResponse struct {
-	Chats         []*Chat                     `json:"chats,omitEmpty"`
-	Messages      []*Message                  `json:"messages,omitEmpty"`
-	Contacts      []*Contact                  `json:"contacts,omitEmpty"`
-	Installations []*multidevice.Installation `json:"installations,omitEmpty"`
+	Chats         []*Chat                     `json:"chats,omitempty"`
```
Comment (adambabik) [inline protocol/messenger.go:77]: Potentially breaking change cc @cammellos.

File: protocol/contact.go (hunk 1 above)
Comment (cammellos) [reply protocol/contact.go:61]: This is ok, I believe, tributeToTalk is not used

File: protocol/message.go (hunk 2 above)
Comment (cammellos) [reply protocol/message.go:96]: This is also likely ok

File: protocol/message.go (hunk 3 above)
Comment (cammellos) [reply protocol/message.go:135]: good to know, I obviously sprinkled the codebase with the wrong version :)

File: protocol/messenger.go (hunk 4 above)
Comment (cammellos) [reply protocol/messenger.go:77]: This is also fine

**Review by cammellos — APPROVED** [review]

**Pull request merged**